### PR TITLE
program / cli / js / js-legacy: Port from SPL

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

It's time to move over the SPL program implementations to their specific repos, but it hasn't been done.

#### Summary of changes

Following the PR for the feature proposal program at https://github.com/solana-program/feature-proposal/pull/1, but for SPL. I ran:

```
# add the remote for the source repo
git remote add spl https://github.com/solana-labs/solana-program-library.git
# fetch the master from spl under a different local branch
git fetch spl master:fromspl
git switch fromspl
# be safe, remove the remote
git remote rm spl
# filter the stake related files
python3 ../../git-filter-repo \
  --path single-pool/program \
  --path single-pool/cli \
  --path single-pool/js/packages/modern \
  --path single-pool/js/packages/classic \
  --path single-pool/README.md \
  --path-rename single-pool/program:program \
  --path-rename single-pool/cli:clients/cli \
  --path-rename single-pool/js/packages/classic:clients/js-legacy \
  --path-rename single-pool/js/packages/modern:clients/js \
  --path-rename single-pool/README.md:README.md \
  --force
# re-add this repo
git remote add origin git@github.com:joncinque/single-pool.git
git fetch origin main
git switch main
git switch fromspl
git rebase main
```

### CODE REVIEW

File: clients/js-legacy/ts-fixup.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+../../ts-fixup.sh
```
Comment (febo) [inline clients/js-legacy/ts-fixup.sh:1]: I think we are missing this linked file.

File: clients/js-legacy/tsconfig-base.json [hunk 2]
```diff
@@ -0,0 +1 @@
+../../tsconfig-base.json
```
Comment (febo) [inline clients/js-legacy/tsconfig-base.json:1]: Same here, missing this linked file.

File: clients/js-legacy/tsconfig-cjs.json [hunk 3]
```diff
@@ -0,0 +1 @@
+../../tsconfig-cjs.json
```
Comment (febo) [inline clients/js-legacy/tsconfig-cjs.json:1]: Same here, missing this linked file.

**Comment by joncinque** [comment]
Thanks for catching that! I added them in #2, and actually redid the legacy JS build to use tsup because the tests only succeeded with Node v20.5, and no other version of Node. I'll push this branch to main so #2 can fix all that

**Pull request closed without merging**